Dockerfile

```diff
@@ -9,16 +9,38 @@ COPY package.json ./
 RUN npm ci --production
 
 # Build h js/css.
-COPY gulpfile.js ./ 
+COPY gulpfile.js ./
 COPY scripts/gulp ./scripts/gulp
 COPY h/static ./h/static
 RUN npm run build
 
-# Stage 2: Build the rest of the app using the build output from Stage 1.
+# Stage 2: Build and install Python dependencies.
+FROM alpine:3.7 AS python-build
```

Comment:
I wonder if it would be better to use a [python2 base image](https://hub.docker.com/_/python/) here rather than raw alpine? Perhaps [2.7.15-alpine3.7](https://github.com/docker-library/python/blob/abcc6175b7a36558f3f805b318a8dd68ce1d18ce/2.7/alpine3.7/Dockerfile).

Reply:
Hmm, a potential problem there is that the Python 2 base image Dockerfile does a bunch of stuff besides just installing the Python 2 development packages. This might affect how the packages are built, which could in turn mean that the packages (esp. the binary ones) don't work correctly when copied to the final image. I'd prefer to stick with keeping the two environments as consistent as possible.

We _could_ use an Alpine Python image as the base image for the main image, but that includes  all the development dependencies, which defeats the point of the Dockerfile logic here / in master to avoid leaving all that stuff unused in the final Docker image.

Reply:
> I don't really see anything that would impact the build but maybe I'm missing something? 

For the pure Python packages, I can't immediately see issues with changing the base image here. For the packages that have binary modules which need to be compiled, the modules are going to be compiled against the header files installed in the python-build container and it is critical that any runtime C/C++ libs used in the main image are compatible with those headers (plus the rest of the build environment). It seems to me that this would be easier to ensure if the same base image is used.

We could _also_ change the main image to use the same Python base image, which would resolve that issue. Is that what you were thinking of? If so, I'd probably make that change separately so we're not trying to deploy too many changes at once.

For the Node build step we don't have this concern because the outputs are JS/CSS/image/font assets which are all portable formats.

Reply:
Yes it looks like the python image is actually slightly less MB's anyway so we should prob just use it instead. It will save build time, size, and make the dockerfile simpler. I'm not sure it makes sense in that case to have another build stage.